Add kwargs to RecordFunctionFast

sraikund16 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123600

Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

sraikund16 committed:
Add kwargs to RecordFunctionFast



Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]

sraikund16 committed:
Update on "Add kwargs to RecordFunctionFast"


Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]

sraikund16 committed:
Update on "Add kwargs to RecordFunctionFast"


Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]

sraikund16 committed:
Update on "Add kwargs to RecordFunctionFast"


Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]

davidberard98 reviewed (approved):
lgtm! had a few nits that I commented on the internal diff.  maybe re-export because it seems like github issues this morning prevented the tests from running fully

sraikund16 committed:
Update on "Add kwargs to RecordFunctionFast"


Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]

sraikund16 committed:
Update on "Add kwargs to RecordFunctionFast"


Differential Revision: [D55897888](https://our.internmc.facebook.com/intern/diff/D55897888/)

[ghstack-poisoned]
